src/netex/read.rs

```diff
@@ -0,0 +1,709 @@
+                    assert!(
+                        assignment_node.name() == "DayTypeAssignment",
+                        "dayTypeAssignments child is expected to be DayTypeAssignment, found {:?}",
+                        assignment_node.name()
+                    );
+                    read_day_type_assignments(collections, context, assignment_node);
+                }
+            }
+            _ => (),
+        };
+    }
+    Ok(())
+}
+
+fn read_day_type_assignments(
+    collections: &mut Collections,
+    context: &mut NetexContext,
+    day_type_assignment: &Element,
+) {
+    let calendar_id = day_type_assignment
+        .get_child("DayTypeRef", &context.namespace)
+        .unwrap()
+        .attr("ref")
+        .unwrap();
+    let day: Date = day_type_assignment
+        .get_child("Date", &context.namespace)
+        .unwrap()
+        .text()
+        .parse()
+        .unwrap();
+    let mut c = collections.calendars.get_mut(calendar_id).unwrap();
+    c.dates.insert(day);
+}
+
+fn read_stop_assignements(
+    _collections: &mut Collections,
+    context: &mut NetexContext,
+    stop_assignments: &Element,
+) {
+    for node in stop_assignments.children() {
+        // assuming all children are PassengerStopAssignment
+        context.routepoint_mapping.insert(
+            node.get_child("ScheduledStopPointRef", &context.namespace)
+                .unwrap()
+                .attr("ref")
+                .unwrap()
+                .to_string(),
+            node.get_child("QuayRef", &context.namespace)
+                .unwrap()
+                .attr("ref")
+                .unwrap()
+                .to_string(),
+        );
+    }
+}
+
+fn read_time_table_frame(
+    collections: &mut Collections,
+    context: &mut NetexContext,
+    time_table_frame: &Element,
+) -> Result<()> {
+    let vj_node = time_table_frame
+        .get_child("vehicleJourneys", &context.namespace)
+        .unwrap();
+    read_vehicle_journeys(collections, context, &vj_node);
+    Ok(())
+}
+
+fn read_organisations(
+    collections: &mut Collections,
+    context: &mut NetexContext,
+    organisations: &Element,
+) -> Result<()> {
+    let companies: Vec<_> = organisations
+        .children()
+        .map(|node| objects::Company {
+            id: node.attr("id").unwrap().to_string(),
+            name: node
+                .get_child("Name", &context.namespace)
+                .unwrap()
+                .text()
+                .to_string(),
+            address: None,
+            url: None,
+            mail: None,
+            phone: None,
+        })
+        .collect();
+    if companies.len() > 0 {
+        context.first_operator_id = companies[0].id.to_string();
+        let mut companies: Vec<objects::Company> = companies
+            .into_iter()
+            .filter(|c| collections.companies.get_idx(&c.id).is_none())
+            .collect();
+        collections.companies.append(&mut companies)?;
```

Comment:
thanks for the hint. It's not possible for `CollectionWithId` because we need a `Result`. An `Err` can occur if an object_id already exists in the collection. 
We could impl it for `Collection`, but we don't need yet the `Extend` trait, and I like having the same `append` method name for both `Collection` and `CollectionWithId`.